tests/test_qutrit_device.py

```diff
@@ -0,0 +1,1022 @@
+        [
+            [1, 2, 2, 2, 10, 1, 1, 5, 1, 1, 1],
+            [(1, 1), (2, 3), (10, 1), (1, 2), (5, 1), (1, 3)],
+            (11, 3),
+            27,
+        ],
+        [[10, 10, 10], [(10, 3)], (3, 3), 30],
+        [[(10, 3)], [(10, 3)], (3, 3), 30],
+    ]
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("shot_list,shot_vector,expected_shape,total_shots", marginal_shot_data)
+    def test_marginal_probs(
+        self, mock_qutrit_device_shots, shot_list, shot_vector, expected_shape, total_shots
+    ):
+        dev = mock_qutrit_device_shots(wires=2, shots=shot_list)
+
+        @qml.qnode(dev)
+        def circuit(x, z):
+            RZ_01 = pnp.array(
+                [
+                    [pnp.exp(-1j * z / 2), 0.0, 0.0],
+                    [0.0, pnp.exp(1j * z / 2), 0.0],
+                    [0.0, 0.0, 1.0],
+                ]
+            )
+
+            c = pnp.cos(x / 2)
+            s = pnp.sin(x / 2) * 1j
+            RX_01 = pnp.array([[c, -s, 0.0], [-s, c, 0.0], [0.0, 0.0, 1.0]])
+
+            qml.QutritUnitary(RZ_01, wires=0)
+            qml.QutritUnitary(RX_01, wires=1)
+            return qml.probs(wires=0)
+
+        res = circuit(0.1, 0.6)
+
+        assert res.shape == expected_shape
+        assert circuit.device._shot_vector == shot_vector
+        assert circuit.device.shots == total_shots
+
+        # test gradient works
+        # TODO: Add after differentiability of qutrit circuits is implemented
+        # res = qml.jacobian(circuit, argnum=[0, 1])(0.1, 0.6)
+
+    shot_data = [
+        [[1, 2, 3, 10], [(1, 1), (2, 1), (3, 1), (10, 1)], (4, 3, 2), 16],
+        [
+            [1, 2, 2, 2, 10, 1, 1, 5, 1, 1, 1],
+            [(1, 1), (2, 3), (10, 1), (1, 2), (5, 1), (1, 3)],
+            (11, 3, 2),
+            27,
+        ],
+        [[10, 10, 10], [(10, 3)], (3, 3, 2), 30],
+        [[(10, 3)], [(10, 3)], (3, 3, 2), 30],
+    ]
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("shot_list,shot_vector,expected_shape,total_shots", shot_data)
+    def test_multiple_probs(
+        self, mock_qutrit_device_shots, shot_list, shot_vector, expected_shape, total_shots
+    ):
+        """Test multiple probability returns"""
+        dev = mock_qutrit_device_shots(wires=2, shots=shot_list)
+
+        @qml.qnode(dev)
+        def circuit(U):
+            qml.QutritUnitary(np.eye(3), wires=0)
+            qml.QutritUnitary(np.eye(3), wires=0)
+            qml.QutritUnitary(U, wires=[0, 1])
+            return qml.probs(wires=0), qml.probs(wires=1)
+
+        res = circuit(pnp.eye(9))
+
+        assert res.shape == expected_shape
+        assert circuit.device._shot_vector == shot_vector
+        assert circuit.device.shots == total_shots
+
+        # test gradient works
+        # TODO: Add after differentiability of qutrit circuits is implemented
+        # res = qml.jacobian(circuit, argnum=[0])(pnp.eye(9, dtype=np.complex128))
+
+
+class TestUnimplemented:
```

Comment:
Mention why these are necessary

Reply:
They're necessary for complete coverage of `QutritDevice`. Some of the methods inherited from `QubitDevice` do not work out of the box for `QutritDevice`, and they have been overriden to raise a `NotImplementedError` when they are called, so I've added these tests to provide coverage for those methods. I will add a comment to the test file as well.